crates/sui-indexer/src/store/module_resolver_v2.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::schema_v2::packages;
+use diesel::ExpressionMethods;
+use diesel::QueryDsl;
+use diesel::RunQueryDsl;
+use move_core_types::language_storage::ModuleId;
+use move_core_types::resolver::ModuleResolver;
+use sui_types::base_types::ObjectID;
+use sui_types::move_package::MovePackage;
+
+use crate::errors::{Context, IndexerError};
+use crate::models_v2::packages::StoredPackage;
+use crate::store::diesel_marco::read_only_blocking;
+use crate::PgConnectionPool;
+
+/// A package resolver that reads packages from the database.
+pub struct IndexerStoreModuleResolver {
+    cp: PgConnectionPool,
+}
```

Comment:
I guess i should have looked at the next PR first where you introduce a bunch of theses caches since you're far ahead of me, i'll leave comments there :)